alacritty_terminal/src/ansi.rs

```diff
@@ -1076,11 +1090,12 @@ where
             },
             ('q', Some(b' ')) => {
                 // DECSCUSR (CSI Ps SP q) -- Set Cursor Style.
-                let style = match arg_or_default!(idx: 0, default: 0) {
+                let ps = arg_or_default!(idx: 0, default: 0);
+                let style = match ps {
                     0 => None,
-                    1 | 2 => Some(CursorStyle::Block),
-                    3 | 4 => Some(CursorStyle::Underline),
-                    5 | 6 => Some(CursorStyle::Beam),
+                    1 | 2 => Some(CursorStyle { shape: CursorShape::Block, blinking: ps == 1 }),
+                    3 | 4 => Some(CursorStyle { shape: CursorShape::Underline, blinking: ps == 3 }),
+                    5 | 6 => Some(CursorStyle { shape: CursorShape::Beam, blinking: ps == 5 }),
```

Comment:
I feel like the additional conditional is just unnecessarily complicated. Expanding this to struct to have separate entries for blinking/non-blinking probably makes it a ton easier to just read without having to think about what is going on.

Reply:
Instead of `1 | 2 => ps == 1`, you'd just do `1 => true, 2 => false`. Highly simplified, but hopefully you get the gist.

Reply:
Something else is wrong around this, since I get the error: "[ERROR] ... Problem with config: missing field \`shape\`; using default" when I insert the following configuration I would expect to work:

```yaml
cursor:
  style:
    blinking: true
```